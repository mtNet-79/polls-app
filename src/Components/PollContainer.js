import PollBox from "./PollBox";

const PollContainer = ({ polls, answered }) => {
  return (
    <>
      {polls.length !== 0 ? (        
          <ul className="flex-row-wrap">
            {polls.map((id) => (
              <li key={id}>
                <PollBox id={id} />
              </li>
            ))}
          </ul>
      ) : (
        <div>
          {answered ? (
            <span className="polls-header center">
              You have answered 0 polls
            </span>
          ) : (
            <span className="polls-header center">
              You have completed all current polls!
            </span>
          )}
        </div>
      )}
    </>
  );
};

export default PollContainer;
