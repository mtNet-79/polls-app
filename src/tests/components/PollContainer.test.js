import { screen, render } from "@testing-library/react";
import PollContainer from "../../components/PollContainer";
import "@testing-library/jest-dom";
import "@testing-library/jest-dom/extend-expect";

const pollsMock = [
  "xj352vofupe1dqz9emx13r",
  "vthrdm985a262al8qx3do",
  "am8ehyc8byjqgar0jgpub9",
  "loxhs1bqm25b708cmbf3g",
  "6ni6ok3ym7mf1p33lnez",
  "8xf0y6ziyjabvozdd253nd",
];

jest.mock("../../components/PollBox", () => () => {
  return <div>Poll Box</div>;
});

//unit tests
test("renders with defualt unaswered polls / snapshot", async () => {
  const { container } = render(
    <PollContainer polls={pollsMock} unanswered={true} />
  );

  

  const itemList = screen.getAllByRole("list");
  const listItems = screen.getAllByRole("listitem");

  expect(itemList.length).toBe(1);
  expect(listItems.length).toBe(6);

  expect(screen.getAllByText(/poll box/i).length).toBe(pollsMock.length);

  expect(container).toMatchSnapshot();
});

test("no answered polls are available", async () => {
  render(<PollContainer polls={[]} answered={false} />);

  const itemList = screen.getByText("You have completed all current polls!");

  expect(itemList).toBeInTheDocument();
});

test("all elements are rendered for defualt answered polls", async () => {
  render(<PollContainer polls={[]} answered={false} />);

  const itemList = screen.queryByRole("list");
  const listItems = screen.queryAllByRole("listitem");

  expect(itemList).not.toBeInTheDocument();
  expect(listItems.length).toBe(0);
});

test("no unanswered polls are available", async () => {
  render(<PollContainer polls={[]} answered={true} />);

  const itemList = screen.getByText("You have answered 0 polls");

  expect(itemList).toBeInTheDocument();
});
