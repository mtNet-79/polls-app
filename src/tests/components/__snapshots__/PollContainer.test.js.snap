// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with defualt unaswered polls / snapshot 1`] = `
<div>
  <ul
    class="flex-row-wrap"
  >
    <li>
      <div>
        Poll Box
      </div>
    </li>
    <li>
      <div>
        Poll Box
      </div>
    </li>
    <li>
      <div>
        Poll Box
      </div>
    </li>
    <li>
      <div>
        Poll Box
      </div>
    </li>
    <li>
      <div>
        Poll Box
      </div>
    </li>
    <li>
      <div>
        Poll Box
      </div>
    </li>
  </ul>
</div>
`;
